### PR TITLE
using PyLong API instead of legacy PyInt API

### LINKED ISSUE #36081: get rid of obsolete PyInt_* and PyInt_*
### Steps To Reproduce

According to https://py3c.readthedocs.io/ all these `PyInt_AS_LONG`, etc should become `PyLong_AS_LONG`, etc etc

However, this is still not fixed, and cause trouble now and then, see e.g. #36080 

In few places, where C/C++ header files are used in Sage, one sees C redefines:
https://github.com/sagemath/sage/blob/0fd596703c824a53f4ffb54556f336de436dde59/src/sage/geometry/triangulation/triangulations.h#L9

But this is not how things should be done in the long run.

### Expected Behavior

There should be no obsolete low level ``PyInt_...` calls.

### Actual Behavior

They are still there.

### Additional Information

_No response_

### Environment

```markdown
not platform-dependent
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by fchapoton**
Fixes #36081 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] I have linked a relevant issue or discussion.

### CODE REVIEW

File: src/sage/symbolic/ginac/numeric.cpp [hunk 1]
```diff
@@ -495,9 +495,9 @@ static PyObject* py_tuple_from_numvector(const std::vector<numeric>& vec)
 ///////////////////////////////////////////////////////////////////////////////
 
 #if PY_MAJOR_VERSION < 3
```
Comment (mkoeppe) [inline src/sage/symbolic/ginac/numeric.cpp]: Surely we can get rid of this?

File: src/sage/symbolic/ginac/numeric.cpp [hunk 2]
```diff
@@ -815,9 +815,9 @@ numeric::numeric(PyObject* o, bool force_py) : basic(&numeric::tinfo_static) {
         if (o == nullptr) py_error("Error");
         if (not force_py) {
 #if PY_MAJOR_VERSION < 3
```
Comment (mkoeppe) [inline src/sage/symbolic/ginac/numeric.cpp]: Remove this `#if` branch

File: src/sage/symbolic/ginac/numeric.cpp [hunk 3]
```diff
@@ -1848,8 +1848,8 @@ const ex numeric::power(const numeric &exponent) const {
         // any PyObjects castable to long are casted
         if (exponent.t == PYOBJECT) {
 #if PY_MAJOR_VERSION < 3
```
Comment (mkoeppe) [inline src/sage/symbolic/ginac/numeric.cpp]: Remove

File: src/sage/symbolic/ginac/numeric.cpp [hunk 4]
```diff
@@ -2486,11 +2486,11 @@ numeric & operator/=(numeric & lh, const numeric & rh)
                 PyObject *p = lh.v._pyobject;
 #if PY_MAJOR_VERSION < 3
```
Comment (mkoeppe) [inline src/sage/symbolic/ginac/numeric.cpp]: Remove

**Comment by mkoeppe** [comment]
please rebase

**Comment by fchapoton** [comment]
no idea if the `map_reduced` doctest failure is related or not

**Comment by fchapoton** [comment]
I cannot reproduce the `map_reduce` doctest failure, which seems to be unrelated.

**Comment by fchapoton** [comment]
all tests are passing now

**Review by mkoeppe — APPROVED** [review]
LGTM. Thanks

**Pull request merged**